packages/@blockprotocol/graph/src/types/subgraph/edges.ts

```diff
@@ -0,0 +1,18 @@
+import { BaseUri } from "@blockprotocol/type-system/slim";
+
+import { EntityId } from "../entity";
+import {
+  KnowledgeGraphOutwardEdge,
+  OntologyOutwardEdge,
+} from "./edges/outward-edge";
+import { Timestamp } from "./time";
+
+export type Edges = {
+  [_: BaseUri]: {
+    [_: number]: OntologyOutwardEdge[];
+  };
+} & {
+  [_: EntityId]: {
+    [_: Timestamp]: KnowledgeGraphOutwardEdge[];
+  };
```

Comment:
We could include further documentation in the key name (not suggesting these exact values, but making them all `_` is missing an opportunity to help understand what's going on)
```suggestion
  [typeBaseUri: BaseUri]: {
    [typeVersion: number]: OntologyOutwardEdge[];
  };
} & {
  [entityBaseId: EntityId]: {
    [edgeFirstCreatedAt: Timestamp]: KnowledgeGraphOutwardEdge[];
  };
```

Reply:
That's a good point! I'm reworking these types a little bit to make them more `tsc` friendly in a follow-up PR so I'll do it then to avoid more rebasing work for the stacked PRs